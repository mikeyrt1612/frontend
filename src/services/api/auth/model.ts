export interface ISigninData {
  token: string,
}
