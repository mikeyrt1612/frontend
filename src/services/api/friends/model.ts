export type IFriendsData = Array<{
  _id: string,
  name: {
    first: string,
    last: string,
    full: string,
  },
  profileImage: string,
}>
