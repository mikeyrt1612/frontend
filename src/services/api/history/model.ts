export enum EActionType {
  CREATED = 'CREATED',
  DELETED = 'DELETED',
  SENT = 'SENT',
  RECEIVED = 'RECEIVED',
}

export type IHistoryData = Array<{
  actionType: EActionType,
  instigator: {
    name: {
      first: string,
      last: string,
      full: string,
    },
  }
  todo: string,
  created: string,
}>
