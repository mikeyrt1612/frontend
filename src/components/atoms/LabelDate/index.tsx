import React, { SFC } from 'react'
import styled from 'styled-components'
import dateFormat from 'dateformat'

export const Label = styled.span`
  color: #ccc;
`

interface IProps {
  date: string,
}

const isToday = (date: string) => {
  const now = new Date()
  const jsDate = new Date(date)
  return now.toDateString() === jsDate.toDateString()
}

const LabelDate: SFC<IProps> = ({ date }) => (
  <Label>
    {
      isToday(date)
      ?
      dateFormat(date, 'HH:MM')
      :
      dateFormat(date, 'mmm d')
    }
  </Label>
)

LabelDate.displayName = 'LabelDate'

export default LabelDate
