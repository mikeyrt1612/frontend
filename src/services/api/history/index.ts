import axios, { AxiosPromise } from 'axios'

import { IHistoryData } from './model'

export const getHistory = (): AxiosPromise<IHistoryData> =>
  axios.get('/api/history')
