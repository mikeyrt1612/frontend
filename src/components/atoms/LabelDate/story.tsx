import React from 'react'
import { storiesOf } from '@storybook/react'

import LabelDate from '.'

storiesOf('atoms/LabelDate', module)
  .add('today', () => (
    <LabelDate date={(new Date()).toISOString()} />
  ))
  .add('before today', () => (
    <LabelDate date="1995-12-17T03:24:00" />
  ))
