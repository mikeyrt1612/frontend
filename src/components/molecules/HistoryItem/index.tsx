import React, { Fragment, SFC } from 'react'
import { Icon, Segment, IconProps } from 'semantic-ui-react'
import styled from 'styled-components'

import LabelPrimary from '@components/atoms/LabelPrimary'
import LabelDate from '@components/atoms/LabelDate'
import { EActionType } from '@services/api/history/model'

export const StatementContainer = styled.span`
  margin-left: 15px !important;
  margin-right: 15px !important;
`

const iconProps: IconProps = {
  [EActionType.SENT]: {
    name: 'send',
  },
  [EActionType.RECEIVED]: {
    name: 'mail',
  },
  [EActionType.CREATED]: {
    name: 'checkmark',
  },
  [EActionType.DELETED]: {
    name: 'delete',
  },
}

const getHistoryTypeStatement = (type: EActionType, name: string) => {
  switch (type) {
    case EActionType.CREATED:
      return (
        <span>You created a new todo</span>
      )
    case EActionType.DELETED:
      return (
        <span>You deleted a todo</span>
      )
    case EActionType.SENT:
      return (
        <Fragment>
          <span>You sent a todo to </span>
          <LabelPrimary text={name} />
        </Fragment>
      )
    case EActionType.RECEIVED:
      return (
        <Fragment>
          <span>You received a todo from </span>
          <LabelPrimary text={name} />
        </Fragment>
      )
  }
}

interface IProps {
  actionType: EActionType,
  instigatorName: string,
  created: string,
}

const HistoryItem: SFC<IProps> = ({ actionType, instigatorName, created }) => (
  <Segment compact>
    <Icon {...iconProps[actionType]} size="large" />
    <StatementContainer>
      {getHistoryTypeStatement(actionType, instigatorName)}
    </StatementContainer>
    <LabelDate date={created} />
  </Segment>
)

HistoryItem.displayName = 'HistoryItem'

export default HistoryItem
