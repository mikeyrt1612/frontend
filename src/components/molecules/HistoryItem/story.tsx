import React from 'react'
import { storiesOf } from '@storybook/react'

import HistoryItem from '.'
import { EActionType } from '@services/api/history/model'

storiesOf('molecules/HistoryItem', module)
.add('sent', () => (
  <HistoryItem
    actionType={EActionType.SENT}
    instigatorName="Michael Thomas"
    created="1995-12-17T03:24:00"
  />
))
.add('received', () => (
  <HistoryItem
    actionType={EActionType.RECEIVED}
    instigatorName="Michael Thomas"
    created="1995-12-17T03:24:00"
  />
))
  .add('created', () => (
    <HistoryItem
      actionType={EActionType.CREATED}
      instigatorName=""
      created="1995-12-17T03:24:00"
    />
  ))
  .add('deleted', () => (
    <HistoryItem
      actionType={EActionType.DELETED}
      instigatorName=""
      created="1995-12-17T03:24:00"
    />
  ))
