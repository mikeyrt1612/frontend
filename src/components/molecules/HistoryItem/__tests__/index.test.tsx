import React, { Fragment, cloneElement } from 'react'
import { shallow } from 'enzyme'
import { Icon } from 'semantic-ui-react'
import renderer from 'react-test-renderer'

import HistoryItem, { StatementContainer } from '../'
import LabelDate from '@components/atoms/LabelDate'
import { EActionType } from '@services/api/history/model'

const mockDate = (new Date()).toISOString()

const getHistoryItemWrapper = (actionType: EActionType) =>
  shallow(
    <HistoryItem
      actionType={actionType}
      instigatorName="MOCK NAME"
      created={mockDate}
    />,
  )

const expectStatementToMatchSnapshot = (actionType: EActionType) => {
  const wrapper = getHistoryItemWrapper(actionType)
  const statement = wrapper.find(StatementContainer)
  const keyedStatements = statement.children().map((e, i) =>
    cloneElement(e.getElement(), { key: i }),
  )
  const tree = renderer.create(
    <Fragment>
      {keyedStatements}
    </Fragment>,
  ).toJSON()
  expect(tree).toMatchSnapshot()
}

describe('<HistoryItem />', () => {
  test('should render the date', () => {
    const wrapper = getHistoryItemWrapper(EActionType.CREATED)
    expect(wrapper.find(LabelDate)).toExist()
    expect(wrapper.find(LabelDate)).toHaveProp('date', mockDate)
  })

  describe('created item', () => {
    test('should have the created icon', () => {
      const wrapper = getHistoryItemWrapper(EActionType.CREATED)
      expect(wrapper.find(Icon)).toExist()
      expect(wrapper.find(Icon)).toHaveProp('name', 'checkmark')
    })

    test('should contain the created message', () => {
      expectStatementToMatchSnapshot(EActionType.CREATED)
    })
  })

  describe('deleted item', () => {
    test('should have the deleted icon', () => {
      const wrapper = getHistoryItemWrapper(EActionType.DELETED)
      expect(wrapper.find(Icon)).toExist()
      expect(wrapper.find(Icon)).toHaveProp('name', 'delete')
    })

    test('should contain the deleted message', () => {
      expectStatementToMatchSnapshot(EActionType.DELETED)
    })
  })

  describe('sent item', () => {
    test('should have the sent icon', () => {
      const wrapper = getHistoryItemWrapper(EActionType.SENT)
      expect(wrapper.find(Icon)).toExist()
      expect(wrapper.find(Icon)).toHaveProp('name', 'send')
    })

    test('should contain the sent message', () => {
      expectStatementToMatchSnapshot(EActionType.SENT)
    })
  })

  describe('recieved item', () => {
    test('should have the recieved icon', () => {
      const wrapper = getHistoryItemWrapper(EActionType.RECEIVED)
      expect(wrapper.find(Icon)).toExist()
      expect(wrapper.find(Icon)).toHaveProp('name', 'mail')
    })

    test('should contain the recieved message', () => {
      expectStatementToMatchSnapshot(EActionType.RECEIVED)
    })
  })
})
