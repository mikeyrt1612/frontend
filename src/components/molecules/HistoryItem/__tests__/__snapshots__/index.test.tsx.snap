// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HistoryItem /> created item should contain the created message 1`] = `
<span>
  You created a new todo
</span>
`;

exports[`<HistoryItem /> deleted item should contain the deleted message 1`] = `
<span>
  You deleted a todo
</span>
`;

exports[`<HistoryItem /> recieved item should contain the recieved message 1`] = `
Array [
  <span>
    You received a todo from 
  </span>,
  <span
    className="sc-bdVaJa gzaUvJ"
  >
    MOCK NAME
  </span>,
]
`;

exports[`<HistoryItem /> sent item should contain the sent message 1`] = `
Array [
  <span>
    You sent a todo to 
  </span>,
  <span
    className="sc-bdVaJa gzaUvJ"
  >
    MOCK NAME
  </span>,
]
`;
